portingdb/check_drops.py

```diff
@@ -385,12 +387,37 @@ def characters(self, content):
             self.source_parts.append(content)
 
 
+@contextlib.contextmanager
+def xmlfile(path):
+    path = Path(path)
+    try:
+        if path.suffix == '.gz':
+            fileobj = gzip.open(path, 'r')
+        elif path.suffix == '.zck':
+            import libdnf.utils  # we only import if needed
+            # there is no library yet to read this, we need to unpack on disk
+            tmppath = tempfile.NamedTemporaryFile(delete=False)
+            tmppath.close()
+            libdnf.utils.decompress(str(path), tmppath.name, 0o644, '.zck')
+            fileobj = open(tmppath.name, 'r')
+        else:
+            # uncompressed XML maybe?
+            fileobj = open(path, 'r')
```

Comment:
This looks like we don't know what to do here. Raise instead?

Reply:
I had that for a while. But if the xml file is there it will work and if it is not and xml file, it will raise later when we attempt to parse it. Given the nature of this script (the target audience is me) I assumed that's OK.

Reply:
```
$ echo bahaha > tmp
$ python3 -m portingdb check-drops -f tmp 
Traceback (most recent call last):
  File "/usr/lib64/python3.7/xml/sax/expatreader.py", line 217, in feed
    self._parser.Parse(data, isFinal)
xml.parsers.expat.ExpatError: syntax error: line 1, column 0

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib64/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib64/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File ".../portingdb/portingdb/__main__.py", line 3, in <module>
    main()
  File ".../portingdb/portingdb/cli.py", line 22, in main
    return cli(obj={})
  File ".../portingdb/__venv__/lib64/python3.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File ".../portingdb/__venv__/lib64/python3.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File ".../portingdb/__venv__/lib64/python3.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File ".../portingdb/__venv__/lib64/python3.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File ".../portingdb/__venv__/lib64/python3.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File ".../portingdb/__venv__/lib64/python3.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File ".../portingdb/portingdb/check_drops.py", line 447, in check_drops
    xml.sax.parse(f, handler)
  File "/usr/lib64/python3.7/xml/sax/__init__.py", line 33, in parse
    parser.parse(source)
  File "/usr/lib64/python3.7/xml/sax/expatreader.py", line 111, in parse
    xmlreader.IncrementalParser.parse(self, source)
  File "/usr/lib64/python3.7/xml/sax/xmlreader.py", line 125, in parse
    self.feed(buffer)
  File "/usr/lib64/python3.7/xml/sax/expatreader.py", line 221, in feed
    self._err_handler.fatalError(exc)
  File "/usr/lib64/python3.7/xml/sax/handler.py", line 38, in fatalError
    raise exception
xml.sax._exceptions.SAXParseException: tmp:1:0: syntax error
```

Reply:
Feel free to ignore that nitpick :)